arangopipe/tests/arangopipe_testcases.py

```diff
@@ -1,170 +1,324 @@
+                        "run_id": "0ef73d9edf08487793c77a1742f4033e",\
+                        "model": model_reg["_key"],\
+                        "model-params": model_params,\
+                        "model-perf": model_perf,\
+                        "pipeline" : "Wine-Regression-Pipeline",\
+                        "project": "Wine-Quality-Assessment",\
+                        "deployment_tag": "Wine_Elastic_Net_Regression",\
+                        "tag": "wine regression model test 1"}
+         self.ap.log_run(run_info)
+         return
+    
+    def provision_deployment(self):
+    
+        ret = self.admin.register_deployment("Wine_Elastic_Net_Regression")
+    
+        return
+        
+    
+    def register_featureset(self):
+        
+        fs_info = {"fixed acidity": "float64",\
+                   "volatile acidity": "float64",\
+                   "citric acid": "float64",\
+                   "residual sugar": "float64",\
+                   "chlorides": "float64",\
+                   "free sulfur dioxide": "float64",\
+                   "total sulfur dioxide": "float64",\
+                   "density": "float64",\
+                   "pH": "float64",\
+                   "sulphates": "float64",\
+                   "alcohol": "float64",\
+                   "quality": "int64",\
+                   "name": "wine_no_transformations"
+                   }
+        ds_reg = self.ap.lookup_dataset("wine_dataset")
+        fs_reg = self.ap.register_featureset(fs_info, ds_reg["_key"])
+        return
+    
+    def log_servingperf(self):
+        to_date = datetime.datetime.now()
+        from_date = to_date - datetime.timedelta(days = 30)
+        ex_servingperf = {"rmse": 0.822242, "r2": 0.12678, "mae": 0.62787,\
+                          "from_date": str(from_date), "to_date": str(to_date)}
+        dep_tag = "Wine_Elastic_Net_Regression"
+        user_id = "prvileged user"
+        ret = self.ap.log_serving_perf(ex_servingperf, dep_tag, user_id)
+       
+        return
+    
+    def dataset_shift_positive(self):
+        ds_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "covariate_shift/cal_housing.csv")
+        df = pd.read_csv(ds_path)
+        req_cols = df.columns.tolist()
+        df = df[req_cols]
+        df1 = df.query("lat <= -119")
+        df2 = df.query("lat > -119")
+        rfd = RF_DatasetShiftDetector()
+        score = rfd.detect_dataset_shift(df1, df2)
+        print ("Detaset shift score : ", score)
+        
+        return
 
+    def dataset_shift_negative(self):
+        ds_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "covariate_shift/cal_housing.csv")
+        df = pd.read_csv(ds_path)
+        req_cols = df.columns.tolist()
+        df = df[req_cols]
+        df1 = df.query("lat <= -119")
+        df2 = df1.copy()
+        rfd = RF_DatasetShiftDetector()
+        score = rfd.detect_dataset_shift(df1, df2)
+        print ("Detaset shift score : ", score)
```

Comment:
This test still doesn't test... when would it fail? I would expect some if (delta> threshold) : fail_test